src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -175,9 +191,24 @@ private async Task ProcessQueueAsync()
                 try
                 {
                     var (work, activityId, cancellationToken) = queueItem;
+                    CancellationTokenSource? cancellationTokenSource = null;
                     lspServices = work.LspServices;
 
-                    var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(CancellationToken, cancellationToken);
```

Comment:
I don't follow why this was moved? This is here so that when the request queue itself is shutdown, in progress work is cancelled.

Reply:
The linked token source existed before my change but didn't do anything. From my understanding, there are other cancellation token sources that handle the request queue shutdown,

Reply:
yeah its already linked - see line 112

Reply:
Going to resolve, please let me know if you still have concerns.